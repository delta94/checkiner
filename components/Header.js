import React from 'react';

const Header = ({ avatar, username, onLogout, isLoggingOut }) => {
  return (
    <header className="header">
      <div className="user">
        <img className="user__avatar" src={avatar} />
        <p className="user__name">{username}</p>
      </div>
      <button className="header__logout" disabled={isLoggingOut} onClick={onLogout}>
        <span>Log out</span>
      </button>

      <style jsx>{`
        .header {
          background: rgba(0, 0, 0, 0.7);
          padding: 1rem;
          display: flex;
          align-items: center;
          justify-content: space-between;
        }
        .user {
          display: flex;
          align-items: center;
          flex: 0 0 50%;
        }
        .user__avatar {
          border-radius: 0.4rem;
          margin-right: 1rem;
          width: 4.8rem;
          height: 4.8rem;
        }
        .user__name {
          color: #fff;
          font: 2rem/2.5rem 'Merienda One', serif;
        }
        .header__logout {
          margin: 0;
          padding: 1rem;
          border-radius: 0.4rem;
          border: none;
          cursor: pointer;
          background: rgba(145, 92, 182, 0.7);
          transition: background 0.2s;
        }
        .header__logout:hover {
          background: rgba(145, 92, 182, 0.9);
        }
        .header__logout[disabled] {
          background: rgba(145, 92, 182, 0.3);
          cursor: not-allowed;
        }
        .header__logout[disabled] span {
          opacity: 0.6;
        }
        .header__logout span {
          line-height: 1rem;
          display: flex;
          align-items: center;
          font: 1.4rem sans-serif;
          color: #fff;
        }
        .header__logout span:before {
          content: '';
          margin-right: 0.5rem;
          width: 1.6rem;
          height: 1.6rem;
          background: url('/static/sign-out.svg') center no-repeat;
          background-size: contain;
        }
      `}</style>
    </header>
  );
};

export default Header;
