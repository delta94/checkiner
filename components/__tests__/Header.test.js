/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Header from 'components/Header';

describe('With Enzyme', () => {
  const children = <p>Hello World!</p>;
  it('Header renders log out button', () => {
    const header = shallow(<Header />);

    expect(header.find('.header__logout span').text()).toEqual('Log out');
  });

  it('Header renders user card with passed avatar url and username', () => {
    const avatar = '/avatars/user.png';
    const username = 'userName';
    const header = shallow(<Header username={username} avatar={avatar} />);

    expect(header.find('.user__avatar').prop('src')).toEqual(avatar);
    expect(header.find('.user__name').text()).toEqual(username);
  });
});

describe('With Snapshot testing', () => {
  it('Header rendered correctly', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
