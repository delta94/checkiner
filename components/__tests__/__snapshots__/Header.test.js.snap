// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot testing Header rendered correctly 1`] = `
<header
  className="jsx-651776104 header"
>
  <div
    className="jsx-651776104 user"
  >
    <img
      className="jsx-651776104 user__avatar"
    />
    <p
      className="jsx-651776104 user__name"
    />
  </div>
  <button
    className="jsx-651776104 header__logout"
  >
    <span
      className="jsx-651776104"
    >
      Log out
    </span>
  </button>
</header>
`;
